server/mocked.js

```diff
@@ -1,59 +1,37 @@
 /* eslint-disable */
-import { Module } from 'module';
+const { Module } = require("module");
 
 const moduleLoadOriginal = Module._load;
 
 const mockedAsyncLocalStorage = {
-    AsyncLocalStorage: class AsyncLocalStorage {
-        run(
-            _store,
-            callback,
-            ...args
-        ) {
-            console.log('Calling mocked AsyncLocalStorage.run()');
-            return callback(...args);
-        }
-
-        getStore() {
-            console.log('Calling mocked AsyncLocalStorage.getStore()');
-            return {};
-        }
-
-        // Other methods below should not be called by the Temporal client package
-
-        disable() {
-            /* noop */
-        }
-
-        exit(
-            callback,
-            ...args
-        ) {
-            return callback(...args);
-        }
-
-        enterWith(_store) {
-            /* noop */
-        }
-    },
+  AsyncLocalStorage: class AsyncLocalStorage {
+    run(_store, callback, ...args) {
+      return callback(...args);
+    }
+    getStore() {
+      return {};
+    }
+    // Other methods below should not be called by the Temporal client package
+    disable() {
+      /* noop */
+    }
+    exit(callback, ...args) {
+      return callback(...args);
+    }
+    enterWith(_store) {
+      /* noop */
+    }
+  },
 };
 
 Module._load = function (uri, parent) {
-    if (uri === 'node:async_hooks' || uri === 'async_hooks') {
-        console.log(`async_hooks requested from ${uri}`);
-        console.log('parent filename', parent.filename);
-        if (
-            parent?.filename?.endsWith('@temporalio/client/lib/connection.js')
-        ) {
-            return mockedAsyncLocalStorage;
-        }
+  if (uri === "node:async_hooks" || uri === "async_hooks") {
+    if (parent?.filename?.endsWith("@temporalio/client/lib/connection.js")) {
+      return mockedAsyncLocalStorage;
     }
+  }
 
-    return moduleLoadOriginal(uri, parent);
+  return moduleLoadOriginal(uri, parent);
 };
 
-export * from './test.js';
-
-// (Module as any)._load = moduleLoadOriginal;
-/* eslint-enable */
-
+module.exports = require("./test.js");
```

Comment:
It is very important to use `require()` here rather than `export from`. If the former syntax is used, the `export from` statement gets hoisted to the top of the file, so that `./test.js` effectively gets loaded before the mock loader is registered (you can see that by adding some `console.log()` statements at the top level of `mocked.js` and `test.js` in your original code).